src/script/E2EIdentity/E2EIdentityEnrollment.ts

```diff
@@ -117,107 +112,69 @@ export class E2EIHandler extends TypedEventEmitter<Events> {
    * @returns
    */
   public isE2EIEnabled() {
-    return !!this.config;
+    return !!this.#config;
   }
 
+  /** will initialize the e2ei enrollment handler eventually triggering an enrollment flow if the device is a fresh new one */
   public async initialize({discoveryUrl, gracePeriodInSeconds}: E2EIHandlerParams) {
     const gracePeriodInMs = gracePeriodInSeconds * TIME_IN_MILLIS.SECOND;
-    this.config = {
+    this.#config = {
       discoveryUrl,
       gracePeriodInMs,
-      timer: new SnoozableTimer({
-        gracePeriodInMS: gracePeriodInMs,
-        onGracePeriodExpired: () => this.processEnrollmentUponExpiry(),
-        onSnoozeExpired: () => this.processEnrollmentUponExpiry(),
-      }),
     };
 
     await this.coreE2EIService.initialize(discoveryUrl);
 
-    this.emit('initialized', {enrollmentConfig: this.config});
+    if (await this.coreE2EIService.isEnrollmentInProgress()) {
+      // If we have an enrollment in progress, we can just finish it (meaning we are coming back from an idp redirect)
+      await this.enroll();
+    } else if (await isFreshMLSSelfClient()) {
+      // When the user logs in to a new device in an environment that has e2ei enabled, they should be forced to enroll
+      await this.startEnrollment(ModalType.ENROLL, false);
+    }
     return this;
   }
 
-  private async processEnrollmentUponExpiry() {
-    const hasCertificate = await hasActiveCertificate();
-    const enrollmentType = hasCertificate ? ModalType.CERTIFICATE_RENEWAL : ModalType.ENROLL;
-    await this.startEnrollment(enrollmentType);
-  }
-
-  public async attemptEnrollment(): Promise<void> {
-    const hasCertificate = await hasActiveCertificate();
-    if (hasCertificate) {
-      // If the client already has a certificate, we don't need to start the enrollment
-      return;
-    }
-    return this.startEnrollment(ModalType.ENROLL);
-  }
+  /**
+   * Will initiate the timer that will regularly prompt the user to enroll (or to renew the certificate if it is about to expire)
+   * @returns the delay under which the next enrollment/renewal modal will be prompted
+   */
+  public async startTimers() {
```

Comment:
This method is the main change here. 

Those timers are handling every single future enrollment triggers (other than the initial enrollment for new devices)